styx-api-service/src/main/java/com/spotify/styx/api/WorkflowResource.java

```diff
@@ -151,10 +165,13 @@ public WorkflowResource(Storage storage, WorkflowValidator workflowValidator,
     if (payload.isEmpty()) {
       return Response.forStatus(Status.BAD_REQUEST.withReasonPhrase("Missing payload."));
     }
-    final WorkflowConfiguration workflowConfig;
+    WorkflowConfiguration workflowConfig;
     try {
       workflowConfig = OBJECT_MAPPER
           .readValue(payload.get().toByteArray(), WorkflowConfiguration.class);
+      workflowConfig = WorkflowConfigurationBuilder.from(workflowConfig).deploymentTime(clock.instant()).build();
+
```

Comment:
i am not entirely sure if this is the best place to do this because I worry that we are might miss different entrypoints for creating/updating workflows.
Or we could extract these two lines in a different function `readFromJsonWithDefaults`

Reply:
or createWorkflowFromJsonWithDefaults